src/components/Header/UserSettings.ts

```diff
@@ -0,0 +1,188 @@
+        z-index: 1;
+        min-width: 12rem;
+      }
+
+      a {
+        font-size: 0.875rem;
+        letter-spacing: 0.01rem;
+        word-spacing: 0.025rem;
+        font-style: normal;
+        font-weight: 400;
+        line-height: 2.5;
+        padding: 0 1.5rem;
+        display: flex;
+        align-items: center;
+        gap: 0.25rem;
+        height: 100%;
+        width: 100%;
+        text-decoration: none;
+        color: var(--bkd-func-fg-black);
+      }
+
+      a img {
+        margin-left: -5.25px;
+      }
+
+      a:hover {
+        color: var(--bkd-brand-red);
+        background: var(--bkd-brand-light-sand);
+        border-left: 6px solid var(--bkd-brand-red);
+        font-weight: 700;
+        padding: 0 calc(1.5rem - 6px);
+      }
+    `,
+  ];
+
+  profile() {
+    return html`<a href="#">${msg("Mein Profil")}</a>`;
+  }
+
+  settings() {
+    return html`<a href="#">${msg("Einstellungen")}</a>`;
+  }
+
+  videos() {
+    const playlist =
+      this.currentLocale === "de"
+        ? "PLLDtLiOuctbx-_EQWgWqTO1MRbX845OEf"
+        : "PLLDtLiOuctbyEegnquAkaW4u8cm62lFAU";
+
+    return html`<a
+      href=${`https://www.youtube.com/playlist?list=${playlist}`}
+      target="_blank"
+      ><img src=${"/icons/external-link.svg"} alt="" width="24" height="24" />
+      ${msg("Video-Tutorials")}</a
+    >`;
+  }
+
+  logout() {
+    return html`<a href="#"
+      ><img src=${"/icons/logout.svg"} alt="" width="24" height="24" />${msg(
+        "Logout"
+      )}</a
+    >`;
+  }
+
+  render() {
+    return html`
+      <button
+        type="button"
+        @click=${() => this.toggle()}
+        aria-label=${msg("Menü Benutzereinstellungen")}
+      >
+        <img src=${"/icons/settings.svg"} alt="" width="32" height="32" />
+      </button>
+      <ul ?hidden=${!this.open}>
+        ${html`${[
+          this.profile(),
+          this.settings(),
+          this.videos(),
+          this.logout(),
+        ].map((link) => html`<li>${link}</li>`)} `}
+      </ul>
+    `;
+  }
+
+  private toggle() {
+    this.open = !this.open;
+    if (this.open) {
+      document.addEventListener("keydown", this.handleKeydown);
+      document.addEventListener("click", this.handleClick);
+    } else {
+      document.removeEventListener("keydown", this.handleKeydown);
+      document.removeEventListener("click", this.handleClick);
+    }
+  }
+
+  private handleClick = (e: MouseEvent) => {
+    const target = e.composedPath()[0];
+    if (
+      target !== this.menuButton &&
+      !this.menuButton.contains(target) &&
+      this.open
+    ) {
+      this.toggle();
+    }
+  };
+
+  private activeLinkId(): number {
+    const active = this.shadowRoot?.activeElement;
+    return [...this.menuLinks].indexOf(active);
+  }
+
+  private nextLinkId(offset: number, start: number): number {
+    const activeId = this.activeLinkId();
+    return activeId + offset < 0
+      ? start
+      : (activeId + offset) % this.menuLinks.length;
```

Comment:
Im Moment hast du den `start` hardcoded (`this.nextLinkId(-1, 3)`), was recht unschön ist.

Vielleicht wäre sowas besser? Oder du hast noch eine schönere Variante?

```
  private activeLinkIndex(): number | null {
    const active = this.shadowRoot?.activeElement;
    const index = [...this.menuLinks].indexOf(active);
    return index === -1 ? null : index;
  }

  private nextLinkIndex(offset: number): number {
    const active = this.activeLinkId();
    const first = 0;
    const last = this.menuLinks.length -1;

    if (active === null) {
      return offset > 0 ? first : last;
    }

    let next = active + offset;
    if (next > last) return first;
    if (next < first) return last;
    return next;
```

Reply:
Nein, ich finde deinen Vorschlag gut.